force-app/main/default/classes/LoyaltyAdapter.cls

```diff
@@ -0,0 +1,204 @@
+        commercepayments.AuthorizationResponse authResponse = new commercepayments.AuthorizationResponse();
+        commercepayments.PaymentMethodTokenizationResponse paymentMethodTokenizationResponse = new commercepayments.PaymentMethodTokenizationResponse();
+        if(authRequest.amount!=null )
+        {
+            authResponse.setAmount(authRequest.amount);
+        }
+        else
+        {
+            throw new SalesforceValidationException('Required Field Missing : Amount');
+        }
+
+        authResponse.setGatewayResultCode('00');
+        authResponse.setGatewayResultCodeDescription('Transaction Normal');
+        authResponse.setGatewayAuthCode('SF'+getRandomNumber(6));
+        authResponse.setGatewayReferenceNumber(getRandomNumber(10));
+        authResponse.setSalesforceResultCodeInfo(SUCCESS_SALESFORCE_RESULT_CODE_INFO);
+        authResponse.setGatewayDate(system.now());
+        paymentMethodTokenizationResponse.setGatewayToken(getRandomNumber(8)+getRandomNumber(8));
+        authResponse.setPaymentMethodTokenizationResponse(paymentMethodTokenizationResponse);
+        return authResponse;
+    }
+
+    public commercepayments.GatewayResponse createAuthReversalResponse(commercepayments.AuthorizationReversalRequest authReversalRequest) {
+        commercepayments.AuthorizationReversalResponse authReversalResponse = new commercepayments.AuthorizationReversalResponse();
+        if(authReversalRequest.amount!=null )
+        {
+            authReversalResponse.setAmount(authReversalRequest.amount);
+        }
+        else
+        {
+            throw new SalesforceValidationException('Required Field Missing : Amount');             
+        }
+
+        system.debug('Response - success');
+        authReversalResponse.setGatewayDate(system.now());
+        authReversalResponse.setGatewayResultCode('00');
+        authReversalResponse.setGatewayResultCodeDescription('Transaction Normal');
+        authReversalResponse.setGatewayReferenceNumber('SF'+getRandomNumber(6));
+        authReversalResponse.setSalesforceResultCodeInfo(SUCCESS_SALESFORCE_RESULT_CODE_INFO);
+        return authReversalResponse;
+    }
+
+    public commercepayments.GatewayResponse createCaptureResponse(commercepayments.CaptureRequest captureRequest) {
+        commercepayments.CaptureResponse captureResponse = new commercepayments.CaptureResponse();
+      
+        PaymentAuthorization paymentAuthorization;
+        PaymentGateway paymentGateway = [SELECT Id, PaymentGatewayName 
+                                        FROM PaymentGateway 
+                                        WHERE PaymentGatewayName = 'Loyalty' LIMIT 1];
+        
+
+        if (paymentGateway != null) {
+            paymentAuthorization = [SELECT GatewayRefNumber 
+                                    FROM PaymentAuthorization 
+                                    WHERE PaymentGatewayId =: paymentGateway.Id];
```

Comment:
Didn't you have a problem with multiple results on this query and should change the filter to the paymentAuthorizationId?